tests/compile-fail/let_return.rs

```diff
@@ -0,0 +1,14 @@
+#![feature(plugin)]
+#![plugin(clippy)]
+
+#![deny(let_and_return)]
+
+fn test() -> i32 {
+    let _y = 0;
+    let x = 5;
+    x            //~ERROR
+}
+
```

Comment:
Add some false-positive guards here too (cases which shouldn't trigger the lint)